lib/tasks/world_taxon_update.rake

```diff
@@ -0,0 +1,156 @@
+
+  task :add_country_name_to_title, %i[log_file_path] => :environment do |_, args|
+    log_file = nil
+    if args[:log_file_path]
+      log_file = File.open(args[:log_file_path], "w")
+      log_file.puts("Updating each worldwide taxon to include country name in their title")
+    end
+    puts("Updating each worldwide taxon to include country name in their title")
+
+    total_taxon_updates = 0
+
+    # Build a taxonomy tree with the grandparent
+    # (common ancestor e.g. /world/all - Help and services around the world) as the root
+    taxonomy = Taxonomy::ExpandedTaxonomy.new(ROOT_CONTENT_ID).build.child_expansion
+    log_file&.puts("Taxonomy has size #{taxonomy.tree.size}")
+    puts("Taxonomy has size #{taxonomy.tree.size}")
+
+    taxonomy.tree.each do |linked_item|
+      # Tree includes root (world/all) - skip that or if the taxon is a GENERIC (template) version
+      next if linked_item.content_id == ROOT_CONTENT_ID || linked_item.internal_name.include?("(GENERIC)")
+
+      internal_name = linked_item.internal_name
+      message = "Internal name = #{internal_name}"
+      log_file ? log_file.puts(message) : puts(message)
+
+      # As this is a tree, we reach all grandchildren without another loop (not a nested array)
+      # example grandchild url /world/passports-and-emergency-travel-documents-cape-verde
+
+      # Skip titles where the country name is already included at the end
+      # e.g. If child - country pages (parent e.g. /world/argentina - UK help and services in Argentina)
+      if internal_name.start_with?("UK help and services in ") || internal_name.start_with?("Living in") || internal_name.start_with?("Travelling to")
+        message = "Skipping #{internal_name} as it already includes the country name"
+        log_file ? log_file.puts(message) : puts(message)
+        next
+      end
+      # Adding the appropriate suffix as follows:
+      # -------------
+      # Birth, death and marriage abroad in COUNTRY_NAME
+      # British embassy or high commission in COUNTRY_NAME
+      # Emergency help for British nationals in COUNTRY_NAME
+      # News and events in COUNTRY_NAME
+      # Passports and emergency travel documents in COUNTRY_NAME
+      # Tax, benefits, pensions and working abroad in COUNTRY_NAME
+      # --------------
+      # Coming to the UK from COUNTRY_NAME
+      # Trade and invest with COUNTRY_NAME
+
+      new_title = ""
+      if internal_name.start_with?("Coming to the UK")
+        message = "adding - ...from COUNTRY_NAME"
+        new_title = internal_name.gsub("(", "from ")
+      elsif internal_name.start_with?("Trade and invest")
+        message = "adding - ...with COUNTRY_NAME"
+        new_title = internal_name.gsub("(", "with ")
+      else
+        message = "adding - ...in COUNTRY_NAME"
+        new_title = internal_name.gsub("(", "in ")
+      end
+      new_title.gsub!(")", "")
+      log_file ? log_file.puts(message) : puts(message)
```

Comment:
Consider moving "new title" building code into separate method, too. It could be something like below, using case statement for more clarity.
```
def create_new_title(internal_name)
  case internal_name
  when ....
```